src/window.c

```diff
@@ -412,6 +426,95 @@ void draw_border(node_t *n, bool focused_node, bool focused_monitor)
 	}
 }
 
+void window_rounded_border(node_t *n)
+{
+	xcb_window_t win = n->id;
+	unsigned int radius = n->client->drawn_border_radius;
```

Comment:
I had that for a bit but the problem was that if `radius` was `0` the border radius wouldn't visualy get updated and because the shape is still set, the window resizing would be broken